src/power_grid_model/validation/validation.py

```diff
@@ -131,8 +135,12 @@ def validate_batch_data(
 
     input_errors: list[ValidationError] = list(validate_unique_ids_across_components(input_data))
 
+    # Convert to row based if in columnar format
+    # TODO(figueroa1395): transform to columnar per single batch once the columnar dataset python extension is finished
```

Comment:
```suggestion
    # TODO(figueroa1395): transform to columnar per single batch scenario once the columnar dataset python extension is finished
```

this is not strictly necessary; just a speed-up

Reply:
Yup, but since everything validation related is done on the Python side, we may as well try to do it as "fast" as possible. Hence, even though this is probably mergable, I will wait a bit since there is no rush.